### PR TITLE
feat: add corner positions to tooltip

### LINKED ISSUE #5483: feat: add corner positions to tooltip
# 🙋 Feature Request

Request to have corner positions added to Tooltip! 

## 🤔 Expected Behavior
Example: A position like `bottom-right` would position the tooltip to the bottom right corner of it's anchor element.

Position suggestions:
`top-left`, `top-right`, `bottom-left`, `bottom-right`
Would also need to consider adding `top-start`, `top-end`, `bottom-start`, `bottom-end` for when RTL direction is on and the tooltip's position would need to be inverted.

### PR DESCRIPTION
**Pull request by SethDonohue**
# Pull Request

## 📖 Description
This adds corner positions to the tooltip to allow for more options.

New positions include:
- top-left
- top-right
- top-start
- top-end

- bottom-left
- bottom-right
- bottom-start
- bottom-end

Fixed Positions:
Screen Shot 2022-01-05 at 2 26 46 PM

Start / End:
Screen Shot 2022-01-05 at 2 27 18 PM

Start / End RTL:
Screen Shot 2022-01-05 at 2 27 30 PM

### 🎫 Issues
closes #5483 

## 👩‍💻 Reviewer Notes

Feedback: Are these location names good?

Please ensure that keyboard functionality, (toggling and Escape actions) work as expected.

### General

- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [x] I have modified an existing component
- [x] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Comment by scomea** [comment]
There's also a pr out that just needs owner approval to enable a center position, so someone's gonna have to do a merge. :-)

**Comment by SethDonohue** [comment]
> There's also a pr out that just needs owner approval to enable a center position, so someone's gonna have to do a merge. :-)
> 
> https://github.com/microsoft/fast/pull/5314/files

Rebased and updated. All looks  good from my perspective.

File: packages/web-components/fast-foundation/src/tooltip/tooltip.ts
Comment (robarbms) [inline packages/web-components/fast-foundation/src/tooltip/tooltip.ts:399]: Should verticalPositioningMode for all of these be "locktodefault" like the other original positions above? Only the default seems to use dynamic positioning.

**Review by robarbms — APPROVED** [review]

File: packages/web-components/fast-foundation/src/tooltip/tooltip.ts
Comment (SethDonohue) [reply packages/web-components/fast-foundation/src/tooltip/tooltip.ts]: removed already defaulted values, thanks!

**Review by eljefe223 — APPROVED** [review]

**Review by scomea — APPROVED** [review]

**Comment by SethDonohue** [comment]
Are any code owners able to review and/or merge? No rush, just pinging before I forget!

File: packages/web-components/fast-components/src/tooltip/tooltip.styles.ts
Comment (EisenbergEffect) [inline packages/web-components/fast-components/src/tooltip/tooltip.styles.ts:103]: I think we should extract the `context.tagFor(AnchoredRegion)` to the top of the function so we only have to make that call once.

File: packages/web-components/fast-foundation/src/tooltip/tooltip.spec.md
Comment (EisenbergEffect) [inline packages/web-components/fast-foundation/src/tooltip/tooltip.spec.md:45]: Thanks for fixing up the spec!

File: packages/web-components/fast-foundation/src/tooltip/tooltip.ts
Comment (EisenbergEffect) [inline packages/web-components/fast-foundation/src/tooltip/tooltip.ts:399]: Seems like everything but the default case uses `locktodefault`. Can we factor that out and reduce duplication?

**Review by EisenbergEffect — CHANGES_REQUESTED** [review]

File: packages/web-components/fast-components/src/tooltip/tooltip.styles.ts
Comment (SethDonohue) [reply packages/web-components/fast-components/src/tooltip/tooltip.styles.ts:103]: Extracted!

**Review by EisenbergEffect — APPROVED** [review]

**Review by chrisdholt — APPROVED** [review]
@SethDonohue looks like you have one merge conflict to resolve. Thanks!

**Comment by SethDonohue** [comment]
@scomea @chrisdholt @nicholasrice I can't bypass the vailed steps to merge this PR, can one of you do so?

**Comment by SethDonohue** [comment]
> @scomea @chrisdholt @nicholasrice I can't bypass the failed steps to merge this PR, can one of you do so?

Hmm on second look, it looks like the api-report is still getting modified, I'll update that.

**Comment by SethDonohue** [comment]
@scomea @chrisdholt @nicholasrice any idea whats up with this linux build error?

**Comment by nicholasrice** [comment]
@awentzel re the code-climate failure.

**Comment by awentzel** [comment]
> @awentzel re the code-climate failure.

Code Climate has an issue currently, I have a filed a ticket and waiting for a response.  If. the failing job is not "required" it can still be merged.  No ETA on solution.

**Comment by awentzel** [comment]
> > @awentzel re the code-climate failure.
> 
> Code Climate has an issue currently, I have a filed a ticket and waiting for a response. If. the failing job is not "required" it can still be merged. No ETA on solution.

I found all tests passed on rebase, thus merged.

**Pull request merged**
